<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="Mutation impact prediction has an important role in understanding the
            roles of mutations in diseases. predatoR package makes prediction about
            amino acid substitutions on protein structures. With using PDB ID, chain ID, residue, 
            reference amino acid, mutated amino acid and gene name (optional), 
            calculates the impact of a mutation  and classifies the mutations
            as pathogenic or neutral.">
<title>Mutation Impact Prediction Based on Network Properties • predatoR</title>
<script src="deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="deps/bootstrap-5.1.3/bootstrap.min.css" rel="stylesheet">
<script src="deps/bootstrap-5.1.3/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="pkgdown.js"></script><meta property="og:title" content="Mutation Impact Prediction Based on Network Properties">
<meta property="og:description" content="Mutation impact prediction has an important role in understanding the
            roles of mutations in diseases. predatoR package makes prediction about
            amino acid substitutions on protein structures. With using PDB ID, chain ID, residue, 
            reference amino acid, mutated amino acid and gene name (optional), 
            calculates the impact of a mutation  and classifies the mutations
            as pathogenic or neutral.">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-dark"><div class="container">
    
    <a class="navbar-brand me-2" href="index.html">predatoR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.0.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="reference/index.html">Reference</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="articles/predatoR_Vignette.html">Introduction to predatoR</a>
  </div>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/berkgurdamar/predatoR/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-home">
<div class="row">
  <main id="main" class="col-md-9"><div class="section level1">
<div class="page-header"><h1 id="-predator-mutation-impact-prediction-based-on-network-properties">
<img src="https://github.com/berkgurdamar/predatoR/blob/main/inst/extdata/predator_logo.png?raw=true" align="left" height="150/"> predatoR: Mutation Impact Prediction Based on Network Properties<a class="anchor" aria-label="anchor" href="#-predator-mutation-impact-prediction-based-on-network-properties"></a>
</h1></div>
<!-- badges: start -->
<!-- <br /> --><!-- <br /> --><!-- # NOTE --><!-- `R-CMD-check` and `Codecov` gave an error due to the Git LFS. Local tests are fine you can download the `predatoR` as described below. -->
</div>
<div class="section level1">
<h1 id="overview">Overview<a class="anchor" aria-label="anchor" href="#overview"></a>
</h1>
<p><code>predatoR</code> is a tool for mutation impact prediction based on network properties.</p>
<p><img src="https://github.com/berkgurdamar/predatoR/blob/main/vignettes/predatoR_workflow.png?raw=true" style="max-width:100%;"></p>
<p><code><a href="reference/predatoR.html">predatoR()</a></code> function is the wrapper function of <code>predatoR</code> package.</p>
<p><code><a href="reference/predatoR.html">predatoR()</a></code> works on each PDB respectively. For each PDB;</p>
<ul>
<li>Downloads/Reads PDB file</li>
<li>Calculates distance between each atom in the structure</li>
<li>Creates a network from PDB</li>
<li>Calculates Degree Centrality Z-Score of each atom</li>
<li>Calculates Eigen Centrality Z-Score of each atom</li>
<li>Calculates Shortest Path Z-Score of each atom</li>
<li>Calculates Betweenness Z-Score of each atom</li>
<li>Calculates Clique Z-Score of each atom</li>
<li>Calculates PageRank Z-Score of each atom</li>
<li>Gets <a href="https://gnomad.broadinstitute.org/" class="external-link">gnomAD</a> Synonymous Z-Score, Missense Z-Score, and pLI Score</li>
<li>Gets BLOSUM62 score of the mutation</li>
<li>Finds the number of <a href="https://www.genome.jp/kegg/" class="external-link">KEGG Pathways</a> which contains the input gene</li>
<li>Gets <a href="http://genic-intolerance.org/" class="external-link">Genic Intolerance</a> Score</li>
<li>Finds the number of GO terms associated with the input gene</li>
<li>Finds the number of diseases associated with the input gene from <a href="https://www.disgenet.org/" class="external-link">DisGeNET</a>
</li>
<li>Gets the Gene Essentiality Score from Online Gene Essentiality (<a href="https://v3.ogee.info/#/home" class="external-link">OGEE</a>) Database</li>
<li>Calculates the median gene expression value of the input gene from <a href="https://gtexportal.org/home/" class="external-link">GTEx</a>
</li>
<li>Calculates 6 different features from Accessible Surface Area and Hydrophobicity Scale of reference and mutant amino acids</li>
<li>Makes prediction</li>
</ul>
</div>
<div class="section level1">
<h1 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h1>
<p>You can install the predatoR via <a href="https://www.r-project.org/nosvn/pandoc/devtools.html" class="external-link">devtools</a>:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://devtools.r-lib.org/" class="external-link">devtools</a></span><span class="op">)</span></span>
<span><span class="fu">install_github</span><span class="op">(</span><span class="st">"berkgurdamar/predatoR"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level1">
<h1 id="usage">Usage<a class="anchor" aria-label="anchor" href="#usage"></a>
</h1>
<p>Mutation impact prediction can be done via <code><a href="reference/predatoR.html">predatoR()</a></code> function:</p>
<p><code><a href="reference/predatoR.html">predatoR()</a></code> uses data.frame structures as an input. data.frame should consist of <strong>‘PDB_ID’</strong>, <strong>‘Chain’</strong>, <strong>‘Position’</strong>, <strong>‘Orig_AA’</strong>, <strong>‘Mut_AA’</strong> and <strong>‘Gene_Name’</strong> (optional). Predictions can be made by using 2 different models, 7 Angstrom (Å)-all atoms model and 5Å-carbon alpha (Cα) atoms only model.</p>
<table class="table">
<thead><tr class="header">
<th align="center">PDB_ID</th>
<th align="center">Chain</th>
<th align="center">Position</th>
<th align="center">Orig_AA</th>
<th align="center">Mut_AA</th>
<th align="center">Gene_Name</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="center">3SQJ</td>
<td align="center">A</td>
<td align="center">196</td>
<td align="center">GLN</td>
<td align="center">LEU</td>
<td align="center">ALB</td>
</tr>
<tr class="even">
<td align="center">3SQJ</td>
<td align="center">A</td>
<td align="center">396</td>
<td align="center">GLU</td>
<td align="center">LYS</td>
<td align="center">ALB</td>
</tr>
</tbody>
</table>
<p><code><a href="reference/predatoR.html">predatoR()</a></code> can work with input which has partially included gene names.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/berkgurdamar/predatoR" class="external-link">predatoR</a></span><span class="op">)</span></span>
<span><span class="co"># Gene name included</span></span>
<span><span class="va">input_df</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">rbind</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"3SQJ"</span>, <span class="st">"A"</span>, <span class="fl">196</span>, <span class="st">"GLN"</span>, <span class="st">"LEU"</span>, <span class="st">"ALB"</span><span class="op">)</span>,</span>
<span>                               <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"3SQJ"</span>, <span class="st">"A"</span>, <span class="fl">396</span>, <span class="st">"GLU"</span>, <span class="st">"LYS"</span>, <span class="st">"ALB"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">pred_res</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/predatoR.html">predatoR</a></span><span class="op">(</span>info_df <span class="op">=</span>  <span class="va">input_df</span>, n_threads <span class="op">=</span> <span class="fl">8</span>, gene_name_info <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co"># Gene name not included</span></span>
<span><span class="va">input_df</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">rbind</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"3SQJ"</span>, <span class="st">"A"</span>, <span class="fl">196</span>, <span class="st">"GLN"</span>, <span class="st">"LEU"</span><span class="op">)</span>,</span>
<span>                               <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"3SQJ"</span>, <span class="st">"A"</span>, <span class="fl">396</span>, <span class="st">"GLU"</span>, <span class="st">"LYS"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">pred_res</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/predatoR.html">predatoR</a></span><span class="op">(</span>info_df <span class="op">=</span>  <span class="va">input_df</span>, n_threads <span class="op">=</span> <span class="fl">8</span>, gene_name_info <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="co"># Partially included gene names</span></span>
<span><span class="va">input_df</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">rbind</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"3SQJ"</span>, <span class="st">"A"</span>, <span class="fl">196</span>, <span class="st">"GLN"</span>, <span class="st">"LEU"</span>, <span class="st">"ALB"</span><span class="op">)</span>,</span>
<span>                               <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"3SQJ"</span>, <span class="st">"A"</span>, <span class="fl">396</span>, <span class="st">"GLU"</span>, <span class="st">"LYS"</span>, <span class="st">""</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">pred_res</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/predatoR.html">predatoR</a></span><span class="op">(</span>info_df <span class="op">=</span>  <span class="va">input_df</span>, n_threads <span class="op">=</span> <span class="fl">8</span>, gene_name_info <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p><code><a href="reference/predatoR.html">predatoR()</a></code> function returns a data.frame which contains additional two columns; <strong>‘Prediction’</strong> and <strong>‘Probability’</strong>. <strong>‘Prediction’</strong> represents the result of the impact prediction and <strong>‘Probability’</strong> represents the probability that the mutation classified as <strong>Pathogenic</strong> or <strong>Neutral</strong>.</p>
<table class="table">
<thead><tr class="header">
<th align="center">PDB_ID</th>
<th align="center">Chain</th>
<th align="center">Position</th>
<th align="center">Orig_AA</th>
<th align="center">Mut_AA</th>
<th align="center">Gene_Name</th>
<th align="center">Prediction</th>
<th align="center">Probability</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="center">3SQJ</td>
<td align="center">A</td>
<td align="center">196</td>
<td align="center">GLN</td>
<td align="center">LEU</td>
<td align="center">ALB</td>
<td align="center">Neutral</td>
<td align="center">0.6521832</td>
</tr>
<tr class="even">
<td align="center">3SQJ</td>
<td align="center">A</td>
<td align="center">396</td>
<td align="center">GLU</td>
<td align="center">LYS</td>
<td align="center">ALB</td>
<td align="center">Neutral</td>
<td align="center">0.6009792</td>
</tr>
</tbody>
</table>
<div class="section level2">
<h2 id="exploratory-usage">Exploratory Usage<a class="anchor" aria-label="anchor" href="#exploratory-usage"></a>
</h2>
<p>Network properties can be calculated by using different distance cutoffs. In this approach, <code><a href="reference/predatoR.html">predatoR()</a></code> does not make any prediction about the mutation, but returns a data frame contains all 24 features annotated to the dataset. Both network formalisation approaches also can be used.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># networks build using all atoms and 7.6Å cutoff</span></span>
<span><span class="va">prediction_result</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/predatoR.html">predatoR</a></span><span class="op">(</span><span class="va">input_df</span>, distance_cutoff <span class="op">=</span> <span class="fl">7.6</span>, network_approach <span class="op">=</span> <span class="st">"all"</span><span class="op">)</span> </span>
<span></span>
<span><span class="co"># networks build using only Cα atoms and 8Å cutoff</span></span>
<span><span class="va">prediction_result</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/predatoR.html">predatoR</a></span><span class="op">(</span><span class="va">input_df</span>, distance_cutoff <span class="op">=</span> <span class="fl">8</span>, network_approach <span class="op">=</span> <span class="st">"ca"</span><span class="op">)</span> </span></code></pre></div>
</div>
<div class="section level2">
<h2 id="utility-functions">Utility Functions<a class="anchor" aria-label="anchor" href="#utility-functions"></a>
</h2>
<p>The wrapper function <code><a href="reference/predatoR.html">predatoR()</a></code> uses the utility functions below;</p>
<ul>
<li><code><a href="reference/read_PDB.html">read_PDB()</a></code></li>
<li><code><a href="reference/PDB2connections.html">PDB2connections()</a></code></li>
<li><code><a href="reference/degree_score.html">degree_score()</a></code></li>
<li><code><a href="reference/eigen_centrality_score.html">eigen_centrality_score()</a></code></li>
<li><code><a href="reference/shorteset_path_score.html">shorteset_path_score()</a></code></li>
<li><code><a href="reference/betweenness_score.html">betweenness_score()</a></code></li>
<li><code><a href="reference/clique_score.html">clique_score()</a></code></li>
<li><code><a href="reference/pagerank_score.html">pagerank_score()</a></code></li>
<li><code><a href="reference/gnomad_scores.html">gnomad_scores()</a></code></li>
<li><code><a href="reference/blosum62_score.html">BLOSUM62_score()</a></code></li>
<li><code><a href="reference/kegg_pathway_number.html">KEGG_pathway_number()</a></code></li>
<li><code><a href="reference/genic_intolerance.html">genic_intolerance()</a></code></li>
<li><code><a href="reference/GO_terms.html">GO_terms()</a></code></li>
<li><code><a href="reference/DisGeNET.html">DisGeNET()</a></code></li>
<li><code><a href="reference/gene_essentiality.html">gene_essentiality()</a></code></li>
<li><code><a href="reference/GTEx.html">GTEx()</a></code></li>
<li><code><a href="reference/amino_acid_features.html">amino_acid_features()</a></code></li>
<li><code><a href="reference/impact_prediction.html">impact_prediction()</a></code></li>
</ul>
<p>Utility functions can be used alone, for more detail please see vignette via <code><a href="articles/predatoR_Vignette.html">vignette("predatoR_Vignette")</a></code>.</p>
</div>
</div>

  </main><aside class="col-md-3"><div class="links">
<h2 data-toc-skip>Links</h2>
<ul class="list-unstyled">
<li><a href="https://github.com/berkgurdamar/predatoR/" class="external-link">Browse source code</a></li>
<li><a href="https://github.com/berkgurdamar/predatoR/issues" class="external-link">Report a bug</a></li>
</ul>
</div>

<div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li><a href="LICENSE.html">Full license</a></li>
<li><small><a href="https://opensource.org/licenses/mit-license.php" class="external-link">MIT</a> + file <a href="LICENSE-text.html">LICENSE</a></small></li>
</ul>
</div>


<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing predatoR</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>Berk Gurdamar <br><small class="roles"> Author, maintainer, copyright holder </small> <a href="https://orcid.org/0000-0003-3811-2211" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
</ul>
</div>

<div class="dev-status">
<h2 data-toc-skip>Dev status</h2>
<ul class="list-unstyled">
<li><a href="https://opensource.org/licenses/MIT" class="external-link"><img src="https://img.shields.io/badge/License-MIT-yellow.svg" alt="License: MIT"></a></li>
<li><a href="https://github.com/berkgurdamar/predatoR/actions/workflows/R-CMD-check.yaml" class="external-link"><img src="https://github.com/berkgurdamar/predatoR/actions/workflows/R-CMD-check.yaml/badge.svg" alt="R-CMD-check"></a></li>
<li><a href="https://app.codecov.io/gh/berkgurdamar/predatoR?branch=main" class="external-link"><img src="https://codecov.io/gh/berkgurdamar/predatoR/branch/main/graph/badge.svg" alt="Codecov test coverage"></a></li>
</ul>
</div>

  </aside>
</div>


    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Berk Gurdamar.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
