<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="predatoR">
<title>Introduction to predatoR • predatoR</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.1.3/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.1.3/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Introduction to predatoR">
<meta property="og:description" content="predatoR">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-dark"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">predatoR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.1.9000</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/predatoR_Vignette.html">Introduction to predatoR</a>
  </div>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/berkgurdamar/predatoR/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">



<script src="predatoR_Vignette_files/htmlwidgets-1.5.4/htmlwidgets.js"></script><link href="predatoR_Vignette_files/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet">
<script src="predatoR_Vignette_files/datatables-binding-0.24/datatables.js"></script><link href="predatoR_Vignette_files/dt-core-1.11.3/css/jquery.dataTables.min.css" rel="stylesheet">
<link href="predatoR_Vignette_files/dt-core-1.11.3/css/jquery.dataTables.extra.css" rel="stylesheet">
<script src="predatoR_Vignette_files/dt-core-1.11.3/js/jquery.dataTables.min.js"></script><link href="predatoR_Vignette_files/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet">
<script src="predatoR_Vignette_files/crosstalk-1.2.0/js/crosstalk.min.js"></script><div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Introduction to predatoR</h1>
                        <h4 data-toc-skip class="author">Berk Gurdamar</h4>
            
            <h4 data-toc-skip class="date">2022-08-18</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/berkgurdamar/predatoR/blob/HEAD/vignettes/predatoR_Vignette.Rmd" class="external-link"><code>vignettes/predatoR_Vignette.Rmd</code></a></small>
      <div class="d-none name"><code>predatoR_Vignette.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="quick-start">Quick Start<a class="anchor" aria-label="anchor" href="#quick-start"></a>
</h2>
<p>In this section, we show you how to use <code>predatoR</code> package for mutation impact prediction.</p>
<p><code>predatoR</code> package makes prediction about impact of a mutation based on an <strong>Adaboost</strong> model and classifies the mutations as <strong>Pathogenic</strong> or <strong>Neutral</strong>. Model was created by using <a href="http://structure.bmc.lu.se/VariBench/" class="external-link">Varibench</a> and <a href="https://www.ncbi.nlm.nih.gov/clinvar/" class="external-link">ClinVar</a> datasets. <a href="http://missense3d.bc.ic.ac.uk/~missense3d/" class="external-link">Missense3D</a> dataset also used for model validation.</p>
<div class="section level4">
<h4 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h4>
<p>You can install the predatoR via <a href="https://www.r-project.org/nosvn/pandoc/devtools.html" class="external-link">devtools</a>:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://devtools.r-lib.org/" class="external-link">devtools</a></span><span class="op">)</span></span>
<span><span class="fu">install_github</span><span class="op">(</span><span class="st">"berkgurdamar/predatoR"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="workflow">Workflow<a class="anchor" aria-label="anchor" href="#workflow"></a>
</h4>
<p><img src="predatoR_workflow.png" style="max-width:100%;"></p>
<p><code><a href="../reference/predatoR.html">predatoR()</a></code> function is the wrapper function of the <code>predatoR</code> package. <code><a href="../reference/predatoR.html">predatoR()</a></code> gets a data.frame structure as an input. Input should contain 5 mandatory and 1 optional columns. Mandatory columns are <strong>‘PDB_ID’</strong>, <strong>‘Chain’</strong>, <strong>‘Position’</strong>, <strong>‘Orig_AA’</strong> and <strong>‘Mut_AA’</strong>. <strong>‘Gene_Name’</strong> column is optional. Predictions can be made by using 2 different models, 7 Angstrom (Å)-all atoms model and 7Å-carbon alpha (Cα) atoms only model.</p>
<p><code><a href="../reference/predatoR.html">predatoR()</a></code> function works on each PDB ID respectively. For each PDB ID;</p>
<ul>
<li>Downloads/Reads PDB file</li>
<li>Calculates distance between each atom in the structure</li>
<li>Creates a network from PDB</li>
<li>Calculates Degree Centrality Z-Score of each atom</li>
<li>Calculates Eigen Centrality Z-Score of each atom</li>
<li>Calculates Shortest Path Z-Score of each atom</li>
<li>Calculates Betweenness Z-Score of each atom</li>
<li>Calculates Clique Z-Score of each atom</li>
<li>Calculates PageRank Z-Score of each atom</li>
<li>Gets <a href="https://gnomad.broadinstitute.org/" class="external-link">gnomAD</a> Synonymous Z-Score, Missense Z-Score, and pLI Score</li>
<li>Gets BLOSUM62 score of the mutation</li>
<li>Finds the number of <a href="https://www.genome.jp/kegg/" class="external-link">KEGG Pathways</a> which contains the input gene</li>
<li>Gets <a href="http://genic-intolerance.org/" class="external-link">Genic Intolerance</a> Score</li>
<li>Finds the number of GO terms associated with the input gene</li>
<li>Finds the number of diseases associated with the input gene from <a href="https://www.disgenet.org/" class="external-link">DisGeNET</a>
</li>
<li>Gets the Gene Essentiality Score from Online Gene Essentiality (<a href="https://v3.ogee.info/#/home" class="external-link">OGEE</a>) Database</li>
<li>Calculates the median gene expression value of the input gene from <a href="https://gtexportal.org/home/" class="external-link">GTEx</a>
</li>
<li>Calculates 6 different features from Accessible Surface Area and Hydrophobicity Scale of reference and mutant amino acids</li>
<li>Makes prediction</li>
</ul>
<p>If <code><a href="../reference/predatoR.html">predatoR()</a></code> couldn’t find any information of any gene such as ‘Genic Intolerance Score’, those inputs will be removed from the query before the prediction step.</p>
</div>
<div class="section level4">
<h4 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a>
</h4>
<p><code><a href="../reference/predatoR.html">predatoR()</a></code> function has 6 arguments;</p>
<ul>
<li>
<strong>info_df:</strong> Input data.frame</li>
<li>
<strong>PDB_path:</strong> PDB file path if the user wants to use PDB structures stored locally (default = NULL)</li>
<li>
<strong>n_threads:</strong> Number of threads (default = NULL)</li>
<li>
<strong>gene_name_info:</strong> Whether there is gene name information in the input or not (default = TRUE)</li>
<li>
<strong>distance_cutoff:</strong> distance cutoff for setting edges (default = 7)</li>
<li>
<strong>network_approach:</strong> network building approach; “all” (default) for using all atoms or “ca” for using Cα atoms only</li>
</ul>
<p>When running the <code><a href="../reference/predatoR.html">predatoR()</a></code> function, if gene name is included in the input, <code>gene_name_info</code> should be set as <code>TRUE</code>. If there is no gene name information in the input (<code>gene_name_info = FALSE</code>), <code><a href="../reference/predatoR.html">predatoR()</a></code> function gets the related gene names from <a href="https://www.ensembl.org/" class="external-link">Ensembl</a> and if there are multiple genes annotated for the PDB ID, choose the gene having the higher gnomAD metrics.</p>
<p><code><a href="../reference/predatoR.html">predatoR()</a></code> can make predictions using only 7Å distance cutoff. For exploratory purposes, different cutoffs can be used and network properties can be calculated.</p>
<p>Example input with the <strong>‘Gene_Name’</strong> information included:</p>
<table class="table">
<thead><tr class="header">
<th align="center">PDB_ID</th>
<th align="center">Chain</th>
<th align="center">Position</th>
<th align="center">Orig_AA</th>
<th align="center">Mut_AA</th>
<th align="center">Gene_Name</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="center">3SQJ</td>
<td align="center">A</td>
<td align="center">196</td>
<td align="center">GLN</td>
<td align="center">LEU</td>
<td align="center">ALB</td>
</tr>
<tr class="even">
<td align="center">3SQJ</td>
<td align="center">A</td>
<td align="center">396</td>
<td align="center">GLU</td>
<td align="center">LYS</td>
<td align="center">ALB</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/berkgurdamar/predatoR" class="external-link">predatoR</a></span><span class="op">)</span></span>
<span><span class="va">prediction_result</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/predatoR.html">predatoR</a></span><span class="op">(</span><span class="va">test_data</span>, gene_name_info <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p>Example input with the <strong>‘Gene_Name’</strong> information does not included:</p>
<table class="table">
<thead><tr class="header">
<th align="center">PDB_ID</th>
<th align="center">Chain</th>
<th align="center">Position</th>
<th align="center">Orig_AA</th>
<th align="center">Mut_AA</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="center">3SQJ</td>
<td align="center">A</td>
<td align="center">196</td>
<td align="center">GLN</td>
<td align="center">LEU</td>
</tr>
<tr class="even">
<td align="center">3SQJ</td>
<td align="center">A</td>
<td align="center">396</td>
<td align="center">GLU</td>
<td align="center">LYS</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">prediction_result</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/predatoR.html">predatoR</a></span><span class="op">(</span><span class="va">test_data</span>, gene_name_info <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<p><code><a href="../reference/predatoR.html">predatoR()</a></code> function can also work with no gene name inputs. If a mutation has no gene name information, it gets the related gene names from <a href="https://www.ensembl.org/" class="external-link">Ensembl</a> and if there are multiple genes annotated for the PDB ID, gets the gene which has maximum GnomAD scores (<code>gene_name_info</code> should set as <code>TRUE</code>).</p>
<table class="table">
<thead><tr class="header">
<th align="center">PDB_ID</th>
<th align="center">Chain</th>
<th align="center">Position</th>
<th align="center">Orig_AA</th>
<th align="center">Mut_AA</th>
<th align="center">Gene_Name</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="center">3SQJ</td>
<td align="center">A</td>
<td align="center">196</td>
<td align="center">GLN</td>
<td align="center">LEU</td>
<td align="center">ALB</td>
</tr>
<tr class="even">
<td align="center">3SQJ</td>
<td align="center">A</td>
<td align="center">396</td>
<td align="center">GLU</td>
<td align="center">LYS</td>
<td align="center">ALB</td>
</tr>
<tr class="odd">
<td align="center">1A4Y</td>
<td align="center">A</td>
<td align="center">169</td>
<td align="center">PRO</td>
<td align="center">LEU</td>
<td align="center"></td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">prediction_result</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/predatoR.html">predatoR</a></span><span class="op">(</span><span class="va">test_data</span>, gene_name_info <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p><code><a href="../reference/predatoR.html">predatoR()</a></code> function uses <code>max_thread_number - 1</code> as a default when <code>n_threads = NULL</code>. <code>n_threads</code> can be set by user.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">prediction_result</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/predatoR.html">predatoR</a></span><span class="op">(</span><span class="va">test_data</span>, n_threads <span class="op">=</span> <span class="fl">8</span><span class="op">)</span> <span class="co"># 8 threads will be used</span></span></code></pre></div>
<p><code><a href="../reference/predatoR.html">predatoR()</a></code> function downloads each PDB by default (<code>PDB_path = NULL</code>) or can use PDB files stored locally when <code>PDB_path</code> specified. It will automatically read and process the input PDB file from the path.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">prediction_result</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/predatoR.html">predatoR</a></span><span class="op">(</span><span class="va">test_data</span>, PDB_path <span class="op">=</span> <span class="st">"path/to/PDB/files/"</span><span class="op">)</span> </span></code></pre></div>
</div>
<div class="section level4">
<h4 id="result">Result<a class="anchor" aria-label="anchor" href="#result"></a>
</h4>
<p><code><a href="../reference/predatoR.html">predatoR()</a></code> function returns a data.frame which contains additional two columns; <strong>‘Prediction’</strong> and <strong>‘Probability’</strong>. <strong>‘Prediction’</strong> represents the result of the impact prediction and <strong>‘Probability’</strong> represents the probability that the mutation classified as <strong>Pathogenic</strong> or <strong>Neutral</strong>.</p>
<table class="table">
<thead><tr class="header">
<th align="center">PDB_ID</th>
<th align="center">Chain</th>
<th align="center">Position</th>
<th align="center">Orig_AA</th>
<th align="center">Mut_AA</th>
<th align="center">Gene_Name</th>
<th align="center">Prediction</th>
<th align="center">Probability</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="center">3SQJ</td>
<td align="center">A</td>
<td align="center">196</td>
<td align="center">GLN</td>
<td align="center">LEU</td>
<td align="center">ALB</td>
<td align="center">Neutral</td>
<td align="center">0.6521832</td>
</tr>
<tr class="even">
<td align="center">3SQJ</td>
<td align="center">A</td>
<td align="center">396</td>
<td align="center">GLU</td>
<td align="center">LYS</td>
<td align="center">ALB</td>
<td align="center">Neutral</td>
<td align="center">0.6009792</td>
</tr>
</tbody>
</table>
</div>
<div class="section level3">
<h3 id="exploratory-usage">Exploratory Usage<a class="anchor" aria-label="anchor" href="#exploratory-usage"></a>
</h3>
<p>Network properties can be calculated by using different distance cutoffs. In this approach, <code><a href="../reference/predatoR.html">predatoR()</a></code> does not make any prediction about the mutation, but returns a data frame contains all 24 features annotated to the dataset. Both network formalisation approaches also can be used.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># networks build using all atoms and 7.6Å cutoff</span></span>
<span><span class="va">prediction_result</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/predatoR.html">predatoR</a></span><span class="op">(</span><span class="va">test_data</span>, distance_cutoff <span class="op">=</span> <span class="fl">7.6</span>, network_approach <span class="op">=</span> <span class="st">"all"</span><span class="op">)</span> </span>
<span></span>
<span><span class="co"># networks build using only Cα atoms and 8Å cutoff</span></span>
<span><span class="va">prediction_result</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/predatoR.html">predatoR</a></span><span class="op">(</span><span class="va">test_data</span>, distance_cutoff <span class="op">=</span> <span class="fl">8</span>, network_approach <span class="op">=</span> <span class="st">"ca"</span><span class="op">)</span> </span></code></pre></div>
</div>
<div class="section level3">
<h3 id="utility-functions">Utility Functions<a class="anchor" aria-label="anchor" href="#utility-functions"></a>
</h3>
<p>The wrapper function <code><a href="../reference/predatoR.html">predatoR()</a></code> uses the utility functions below;</p>
<ul>
<li><code><a href="../reference/read_PDB.html">read_PDB()</a></code></li>
<li><code><a href="../reference/PDB2connections.html">PDB2connections()</a></code></li>
<li><code><a href="../reference/degree_score.html">degree_score()</a></code></li>
<li><code><a href="../reference/eigen_centrality_score.html">eigen_centrality_score()</a></code></li>
<li><code><a href="../reference/shorteset_path_score.html">shorteset_path_score()</a></code></li>
<li><code><a href="../reference/betweenness_score.html">betweenness_score()</a></code></li>
<li><code><a href="../reference/clique_score.html">clique_score()</a></code></li>
<li><code><a href="../reference/pagerank_score.html">pagerank_score()</a></code></li>
<li><code><a href="../reference/gnomad_scores.html">gnomad_scores()</a></code></li>
<li><code><a href="../reference/blosum62_score.html">BLOSUM62_score()</a></code></li>
<li><code><a href="../reference/kegg_pathway_number.html">KEGG_pathway_number()</a></code></li>
<li><code><a href="../reference/genic_intolerance.html">genic_intolerance()</a></code></li>
<li><code><a href="../reference/GO_terms.html">GO_terms()</a></code></li>
<li><code><a href="../reference/DisGeNET.html">DisGeNET()</a></code></li>
<li><code><a href="../reference/gene_essentiality.html">gene_essentiality()</a></code></li>
<li><code><a href="../reference/GTEx.html">GTEx()</a></code></li>
<li><code><a href="../reference/amino_acid_features.html">amino_acid_features()</a></code></li>
<li><code><a href="../reference/impact_prediction.html">impact_prediction()</a></code></li>
</ul>
<div class="section level4">
<h4 id="read_pdb">
<code>read_PDB</code><a class="anchor" aria-label="anchor" href="#read_pdb"></a>
</h4>
<p><code><a href="../reference/read_PDB.html">read_PDB()</a></code> function downloads input PDB by default but can read and process locally stored PDB structures. After reading, create a matrix that contains only ATOM labelled structures.</p>
<div class="section level5">
<h5 id="arguments-1">Arguments<a class="anchor" aria-label="anchor" href="#arguments-1"></a>
</h5>
<p><code><a href="../reference/read_PDB.html">read_PDB()</a></code> has 3 arguments;</p>
<ul>
<li>
<strong>PDB_ID:</strong> PDB ID (E.g. “3SQJ”)</li>
<li>
<strong>PDB_path:</strong> PDB file path if files stored locally (default = FALSE)</li>
<li>
<strong>network_approach:</strong> network building approach; “all” (default) for using all atoms or “ca” for using Cα atoms only</li>
</ul>
</div>
<div class="section level5">
<h5 id="usage">Usage<a class="anchor" aria-label="anchor" href="#usage"></a>
</h5>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># if file stored locally</span></span>
<span><span class="va">PDB_2DN2_structure</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/read_PDB.html">read_PDB</a></span><span class="op">(</span><span class="st">"3SQJ"</span>, PDB_path <span class="op">=</span> <span class="st">"path/to/PDB/files/"</span><span class="op">)</span> </span>
<span><span class="co"># if file is going to be downloded</span></span>
<span><span class="va">PDB_2DN2_structure</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/read_PDB.html">read_PDB</a></span><span class="op">(</span><span class="st">"3SQJ"</span><span class="op">)</span> </span></code></pre></div>
</div>
</div>
<div class="section level4">
<h4 id="pdb2connections">
<code>PDB2connections</code><a class="anchor" aria-label="anchor" href="#pdb2connections"></a>
</h4>
<p><code><a href="../reference/PDB2connections.html">PDB2connections()</a></code> function calculates distances between each ATOM in the PDB structure and creates an edge between atoms if the distance &lt;= 7Å. As a result, returns a list contains data frames of different chains interactions.</p>
<div class="section level5">
<h5 id="arguments-2">Arguments<a class="anchor" aria-label="anchor" href="#arguments-2"></a>
</h5>
<p><code><a href="../reference/PDB2connections.html">PDB2connections()</a></code> has 5 arguments;</p>
<ul>
<li>
<strong>atom_matrix:</strong> Matrix created by <code><a href="../reference/read_PDB.html">read_PDB()</a></code> function</li>
<li>
<strong>filtered_info_df:</strong> data.frame contains only one PDB entries</li>
<li>
<strong>n_threads:</strong> number of threads (default = NULL)</li>
<li>
<strong>single_run:</strong> should be set as TRUE when using <code><a href="../reference/PDB2connections.html">PDB2connections()</a></code> function alone (default = TRUE)</li>
<li>
<strong>distance_cutoff:</strong> distance cutoff for setting edges (default = 7)</li>
</ul>
</div>
<div class="section level5">
<h5 id="usage-1">Usage<a class="anchor" aria-label="anchor" href="#usage-1"></a>
</h5>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">test_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">rbind</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"3SQJ"</span>, <span class="st">"A"</span>, <span class="fl">3</span>, <span class="st">"HIS"</span>, <span class="st">"GLN"</span>, <span class="st">"ALB"</span><span class="op">)</span>,</span>
<span>                                 <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"3SQJ"</span>, <span class="st">"A"</span>, <span class="fl">60</span>, <span class="st">"GLU"</span>, <span class="st">"LYS"</span>,<span class="st">"ALB"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">test_data</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"PDB_ID"</span>, <span class="st">"Chain"</span>, <span class="st">"Position"</span>, <span class="st">"Orig_AA"</span>, <span class="st">"Mut_AA"</span>, <span class="st">"Gene_Name"</span><span class="op">)</span></span>
<span><span class="va">edge_list</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/PDB2connections.html">PDB2connections</a></span><span class="op">(</span>atom_matrix <span class="op">=</span> <span class="va">PDB_2DN2_structure</span>, filtered_info_df <span class="op">=</span> <span class="va">test_data</span>, n_threads <span class="op">=</span> <span class="fl">8</span>, single_run <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> </span></code></pre></div>
</div>
</div>
<div class="section level4">
<h4 id="degree_score">
<code>degree_score</code><a class="anchor" aria-label="anchor" href="#degree_score"></a>
</h4>
<p><code><a href="../reference/degree_score.html">degree_score()</a></code> function calculates the total number of connections of all nodes. Calculates Z-scores of each node and returns scores of desired nodes.</p>
<div class="section level5">
<h5 id="arguments-3">Arguments<a class="anchor" aria-label="anchor" href="#arguments-3"></a>
</h5>
<p><code><a href="../reference/degree_score.html">degree_score()</a></code> has 2 arguments;</p>
<ul>
<li>
<strong>edge_list:</strong> List created by <code><a href="../reference/PDB2connections.html">PDB2connections()</a></code> function</li>
<li>
<strong>filtered_info_df:</strong> data.frame contains only one PDB entries</li>
</ul>
</div>
<div class="section level5">
<h5 id="usage-2">Usage<a class="anchor" aria-label="anchor" href="#usage-2"></a>
</h5>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">test_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">rbind</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"3SQJ"</span>, <span class="st">"A"</span>, <span class="fl">3</span>, <span class="st">"HIS"</span>, <span class="st">"GLN"</span>, <span class="st">"ALB"</span><span class="op">)</span>,</span>
<span>                                 <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"3SQJ"</span>, <span class="st">"A"</span>, <span class="fl">60</span>, <span class="st">"GLU"</span>, <span class="st">"LYS"</span>,<span class="st">"ALB"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">test_data</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"PDB_ID"</span>, <span class="st">"Chain"</span>, <span class="st">"Position"</span>, <span class="st">"Orig_AA"</span>, <span class="st">"Mut_AA"</span>, <span class="st">"Gene_Name"</span><span class="op">)</span></span>
<span><span class="va">degree_z_score</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/degree_score.html">degree_score</a></span><span class="op">(</span>edge_list <span class="op">=</span> <span class="va">edge_list</span>, filtered_info_df <span class="op">=</span> <span class="va">test_data</span><span class="op">)</span> </span></code></pre></div>
</div>
</div>
<div class="section level4">
<h4 id="eigen_centrality_score">
<code>eigen_centrality_score</code><a class="anchor" aria-label="anchor" href="#eigen_centrality_score"></a>
</h4>
<p><code><a href="../reference/eigen_centrality_score.html">eigen_centrality_score()</a></code> function calculates the total number of connections of nodes to which a node is connected. Calculates Z-scores of all nodes and returns scores of desired nodes.</p>
<div class="section level5">
<h5 id="arguments-4">Arguments<a class="anchor" aria-label="anchor" href="#arguments-4"></a>
</h5>
<p><code><a href="../reference/eigen_centrality_score.html">eigen_centrality_score()</a></code> has 2 arguments;</p>
<ul>
<li>
<strong>edge_list:</strong> List created by <code><a href="../reference/PDB2connections.html">PDB2connections()</a></code> function</li>
<li>
<strong>filtered_info_df:</strong> data.frame contains only one PDB entries</li>
</ul>
</div>
<div class="section level5">
<h5 id="usage-3">Usage<a class="anchor" aria-label="anchor" href="#usage-3"></a>
</h5>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">test_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">rbind</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"3SQJ"</span>, <span class="st">"A"</span>, <span class="fl">3</span>, <span class="st">"HIS"</span>, <span class="st">"GLN"</span>, <span class="st">"ALB"</span><span class="op">)</span>,</span>
<span>                                 <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"3SQJ"</span>, <span class="st">"A"</span>, <span class="fl">60</span>, <span class="st">"GLU"</span>, <span class="st">"LYS"</span>,<span class="st">"ALB"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">test_data</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"PDB_ID"</span>, <span class="st">"Chain"</span>, <span class="st">"Position"</span>, <span class="st">"Orig_AA"</span>, <span class="st">"Mut_AA"</span>, <span class="st">"Gene_Name"</span><span class="op">)</span></span>
<span><span class="va">eigen_centrality_z_score</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/eigen_centrality_score.html">eigen_centrality_score</a></span><span class="op">(</span>edge_list <span class="op">=</span> <span class="va">edge_list</span>, filtered_info_df <span class="op">=</span> <span class="va">test_data</span><span class="op">)</span> </span></code></pre></div>
</div>
</div>
<div class="section level4">
<h4 id="shorteset_path_score">
<code>shorteset_path_score</code><a class="anchor" aria-label="anchor" href="#shorteset_path_score"></a>
</h4>
<p><code><a href="../reference/shorteset_path_score.html">shorteset_path_score()</a></code> function calculates the distances between each node via <code>shortest.paths()</code> function of igraph package, sum distances of each nodes, calculates Z-scores and returns scores of desired nodes.</p>
<div class="section level5">
<h5 id="arguments-5">Arguments<a class="anchor" aria-label="anchor" href="#arguments-5"></a>
</h5>
<p><code><a href="../reference/shorteset_path_score.html">shorteset_path_score()</a></code> has 2 arguments;</p>
<ul>
<li>
<strong>edge_list:</strong> List created by <code><a href="../reference/PDB2connections.html">PDB2connections()</a></code> function</li>
<li>
<strong>filtered_info_df:</strong> data.frame contains only one PDB entries</li>
</ul>
</div>
<div class="section level5">
<h5 id="usage-4">Usage<a class="anchor" aria-label="anchor" href="#usage-4"></a>
</h5>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">test_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">rbind</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"3SQJ"</span>, <span class="st">"A"</span>, <span class="fl">3</span>, <span class="st">"HIS"</span>, <span class="st">"GLN"</span>, <span class="st">"ALB"</span><span class="op">)</span>,</span>
<span>                                 <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"3SQJ"</span>, <span class="st">"A"</span>, <span class="fl">60</span>, <span class="st">"GLU"</span>, <span class="st">"LYS"</span>,<span class="st">"ALB"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">test_data</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"PDB_ID"</span>, <span class="st">"Chain"</span>, <span class="st">"Position"</span>, <span class="st">"Orig_AA"</span>, <span class="st">"Mut_AA"</span>, <span class="st">"Gene_Name"</span><span class="op">)</span></span>
<span><span class="va">shorteset_path_z_score</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/shorteset_path_score.html">shorteset_path_score</a></span><span class="op">(</span>edge_list <span class="op">=</span> <span class="va">edge_list</span>, filtered_info_df <span class="op">=</span> <span class="va">test_data</span><span class="op">)</span> </span></code></pre></div>
</div>
</div>
<div class="section level4">
<h4 id="betweenness_score">
<code>betweenness_score</code><a class="anchor" aria-label="anchor" href="#betweenness_score"></a>
</h4>
<p><code><a href="../reference/betweenness_score.html">betweenness_score()</a></code> function calculates Betweenness Score of each node by using <code>betweenness()</code> function of igraph package, turns the scores into Z-scores and returns scores of desired nodes.</p>
<div class="section level5">
<h5 id="arguments-6">Arguments<a class="anchor" aria-label="anchor" href="#arguments-6"></a>
</h5>
<p><code><a href="../reference/betweenness_score.html">betweenness_score()</a></code> has 2 arguments;</p>
<ul>
<li>
<strong>edge_list:</strong> List created by <code><a href="../reference/PDB2connections.html">PDB2connections()</a></code> function</li>
<li>
<strong>filtered_info_df:</strong> data.frame contains only one PDB entries</li>
</ul>
</div>
<div class="section level5">
<h5 id="usage-5">Usage<a class="anchor" aria-label="anchor" href="#usage-5"></a>
</h5>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">test_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">rbind</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"3SQJ"</span>, <span class="st">"A"</span>, <span class="fl">3</span>, <span class="st">"HIS"</span>, <span class="st">"GLN"</span>, <span class="st">"ALB"</span><span class="op">)</span>,</span>
<span>                                 <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"3SQJ"</span>, <span class="st">"A"</span>, <span class="fl">60</span>, <span class="st">"GLU"</span>, <span class="st">"LYS"</span>,<span class="st">"ALB"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">test_data</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"PDB_ID"</span>, <span class="st">"Chain"</span>, <span class="st">"Position"</span>, <span class="st">"Orig_AA"</span>, <span class="st">"Mut_AA"</span>, <span class="st">"Gene_Name"</span><span class="op">)</span></span>
<span><span class="va">betweenness_z_score</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/betweenness_score.html">betweenness_score</a></span><span class="op">(</span>edge_list <span class="op">=</span> <span class="va">edge_list</span>, filtered_info_df <span class="op">=</span> <span class="va">test_data</span><span class="op">)</span> </span></code></pre></div>
</div>
</div>
<div class="section level4">
<h4 id="clique_score">
<code>clique_score</code><a class="anchor" aria-label="anchor" href="#clique_score"></a>
</h4>
<p><code><a href="../reference/clique_score.html">clique_score()</a></code> function calculates Clique score of each atom and turns the desired nodes’ scores into Z-scores.</p>
<div class="section level5">
<h5 id="arguments-7">Arguments<a class="anchor" aria-label="anchor" href="#arguments-7"></a>
</h5>
<p><code><a href="../reference/clique_score.html">clique_score()</a></code> has 4 arguments;</p>
<ul>
<li>
<strong>edge_list:</strong> List created by <code><a href="../reference/PDB2connections.html">PDB2connections()</a></code> function</li>
<li>
<strong>filtered_info_df:</strong> data.frame contains only one PDB entries</li>
<li>
<strong>n_threads:</strong> number of threads (default = NULL)</li>
<li>
<strong>single_run:</strong> should be set as TRUE when using <code><a href="../reference/clique_score.html">clique_score()</a></code> function alone (default = TRUE)</li>
</ul>
</div>
<div class="section level5">
<h5 id="usage-6">Usage<a class="anchor" aria-label="anchor" href="#usage-6"></a>
</h5>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">test_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">rbind</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"3SQJ"</span>, <span class="st">"A"</span>, <span class="fl">3</span>, <span class="st">"HIS"</span>, <span class="st">"GLN"</span>, <span class="st">"ALB"</span><span class="op">)</span>,</span>
<span>                                 <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"3SQJ"</span>, <span class="st">"A"</span>, <span class="fl">60</span>, <span class="st">"GLU"</span>, <span class="st">"LYS"</span>,<span class="st">"ALB"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">test_data</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"PDB_ID"</span>, <span class="st">"Chain"</span>, <span class="st">"Position"</span>, <span class="st">"Orig_AA"</span>, <span class="st">"Mut_AA"</span>, <span class="st">"Gene_Name"</span><span class="op">)</span></span>
<span><span class="va">clique_z_score</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/clique_score.html">clique_score</a></span><span class="op">(</span>edge_list <span class="op">=</span> <span class="va">edge_list</span>, filtered_info_df <span class="op">=</span> <span class="va">test_data</span>, n_threads <span class="op">=</span> <span class="fl">8</span>, single_run <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> </span></code></pre></div>
</div>
</div>
<div class="section level4">
<h4 id="pagerank_score">
<code>pagerank_score</code><a class="anchor" aria-label="anchor" href="#pagerank_score"></a>
</h4>
<p><code><a href="../reference/pagerank_score.html">pagerank_score()</a></code> function assigns an importance score to every node via <code>page_rank()</code> function of igraph package. Turns the scores into Z-scores and returns scores of desired nodes.</p>
<div class="section level5">
<h5 id="arguments-8">Arguments<a class="anchor" aria-label="anchor" href="#arguments-8"></a>
</h5>
<p><code><a href="../reference/pagerank_score.html">pagerank_score()</a></code> has 2 arguments;</p>
<ul>
<li>
<strong>edge_list:</strong> List created by <code><a href="../reference/PDB2connections.html">PDB2connections()</a></code> function</li>
<li>
<strong>filtered_info_df:</strong> data.frame contains only one PDB entries</li>
</ul>
</div>
<div class="section level5">
<h5 id="usage-7">Usage<a class="anchor" aria-label="anchor" href="#usage-7"></a>
</h5>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">test_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">rbind</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"3SQJ"</span>, <span class="st">"A"</span>, <span class="fl">3</span>, <span class="st">"HIS"</span>, <span class="st">"GLN"</span>, <span class="st">"ALB"</span><span class="op">)</span>,</span>
<span>                                 <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"3SQJ"</span>, <span class="st">"A"</span>, <span class="fl">60</span>, <span class="st">"GLU"</span>, <span class="st">"LYS"</span>,<span class="st">"ALB"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">test_data</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"PDB_ID"</span>, <span class="st">"Chain"</span>, <span class="st">"Position"</span>, <span class="st">"Orig_AA"</span>, <span class="st">"Mut_AA"</span>, <span class="st">"Gene_Name"</span><span class="op">)</span></span>
<span><span class="va">pagerank_z_score</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/pagerank_score.html">pagerank_score</a></span><span class="op">(</span>edge_list <span class="op">=</span> <span class="va">edge_list</span>, filtered_info_df <span class="op">=</span> <span class="va">test_data</span><span class="op">)</span> </span></code></pre></div>
</div>
</div>
<div class="section level4">
<h4 id="gnomad_scores">
<code>gnomad_scores</code><a class="anchor" aria-label="anchor" href="#gnomad_scores"></a>
</h4>
<p><code><a href="../reference/gnomad_scores.html">gnomad_scores()</a></code> function gets input gene’s gnomAD Synonymous Z-Score, Missense Z-Score, and PLI Score from data retrieved from <a href="https://gnomad.broadinstitute.org/" class="external-link">gnomAD</a>. If there is no gene name information in the input, <code><a href="../reference/gnomad_scores.html">gnomad_scores()</a></code> finds the genes that related with input PDB chain ID by using a dataset retrieved from <a href="https://www.ensembl.org/" class="external-link">Ensembl</a>. If there are multiple genes annotated for same PDB, <code><a href="../reference/gnomad_scores.html">gnomad_scores()</a></code> function gets the gene that has maximum Synonymous Z-Score, Missense Z-Score, and pLI Score.</p>
<div class="section level5">
<h5 id="arguments-9">Arguments<a class="anchor" aria-label="anchor" href="#arguments-9"></a>
</h5>
<p><code><a href="../reference/gnomad_scores.html">gnomad_scores()</a></code> has 1 arguments;</p>
<ul>
<li>
<strong>filtered_info_df:</strong> data.frame contains only one PDB entries</li>
</ul>
</div>
<div class="section level5">
<h5 id="usage-8">Usage<a class="anchor" aria-label="anchor" href="#usage-8"></a>
</h5>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">test_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">rbind</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"3SQJ"</span>, <span class="st">"A"</span>, <span class="fl">3</span>, <span class="st">"HIS"</span>, <span class="st">"GLN"</span>, <span class="st">"ALB"</span><span class="op">)</span>,</span>
<span>                                 <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"3SQJ"</span>, <span class="st">"A"</span>, <span class="fl">60</span>, <span class="st">"GLU"</span>, <span class="st">"LYS"</span>,<span class="st">"ALB"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">test_data</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"PDB_ID"</span>, <span class="st">"Chain"</span>, <span class="st">"Position"</span>, <span class="st">"Orig_AA"</span>, <span class="st">"Mut_AA"</span>, <span class="st">"Gene_Name"</span><span class="op">)</span></span>
<span><span class="va">gnomad_score</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/gnomad_scores.html">gnomad_scores</a></span><span class="op">(</span>filtered_info_df <span class="op">=</span> <span class="va">test_data</span><span class="op">)</span> </span></code></pre></div>
</div>
</div>
<div class="section level4">
<h4 id="blosum62_score">
<code>BLOSUM62_score</code><a class="anchor" aria-label="anchor" href="#blosum62_score"></a>
</h4>
<p><code><a href="../reference/blosum62_score.html">BLOSUM62_score()</a></code> function returns BLOSUM62 scores of input mutations.</p>
<div class="section level5">
<h5 id="arguments-10">Arguments<a class="anchor" aria-label="anchor" href="#arguments-10"></a>
</h5>
<p><code><a href="../reference/blosum62_score.html">BLOSUM62_score()</a></code> has 1 arguments;</p>
<ul>
<li>
<strong>info_df:</strong> data.frame contains all the input mutations</li>
</ul>
</div>
<div class="section level5">
<h5 id="usage-9">Usage<a class="anchor" aria-label="anchor" href="#usage-9"></a>
</h5>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">test_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">rbind</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"3SQJ"</span>, <span class="st">"A"</span>, <span class="fl">3</span>, <span class="st">"HIS"</span>, <span class="st">"GLN"</span>, <span class="st">"ALB"</span><span class="op">)</span>,</span>
<span>                                 <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"3SQJ"</span>, <span class="st">"A"</span>, <span class="fl">60</span>, <span class="st">"GLU"</span>, <span class="st">"LYS"</span>,<span class="st">"ALB"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">test_data</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"PDB_ID"</span>, <span class="st">"Chain"</span>, <span class="st">"Position"</span>, <span class="st">"Orig_AA"</span>, <span class="st">"Mut_AA"</span>, <span class="st">"Gene_Name"</span><span class="op">)</span></span>
<span><span class="va">BLOSUM62_scores</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/blosum62_score.html">BLOSUM62_score</a></span><span class="op">(</span>info_df <span class="op">=</span> <span class="va">test_data</span><span class="op">)</span> </span></code></pre></div>
</div>
</div>
<div class="section level4">
<h4 id="kegg_pathway_number">
<code>KEGG_pathway_number</code><a class="anchor" aria-label="anchor" href="#kegg_pathway_number"></a>
</h4>
<p><code><a href="../reference/kegg_pathway_number.html">KEGG_pathway_number()</a></code> function finds the number of KEGG Pathways which related with the input genes.</p>
<div class="section level5">
<h5 id="arguments-11">Arguments<a class="anchor" aria-label="anchor" href="#arguments-11"></a>
</h5>
<p><code><a href="../reference/kegg_pathway_number.html">KEGG_pathway_number()</a></code> has 1 arguments;</p>
<ul>
<li>
<strong>filtered_info_df:</strong> input data.frame contains only one PDB entries</li>
</ul>
</div>
<div class="section level5">
<h5 id="usage-10">Usage<a class="anchor" aria-label="anchor" href="#usage-10"></a>
</h5>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">test_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">rbind</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"3SQJ"</span>, <span class="st">"A"</span>, <span class="fl">3</span>, <span class="st">"HIS"</span>, <span class="st">"GLN"</span>, <span class="st">"ALB"</span><span class="op">)</span>,</span>
<span>                                 <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"3SQJ"</span>, <span class="st">"A"</span>, <span class="fl">60</span>, <span class="st">"GLU"</span>, <span class="st">"LYS"</span>,<span class="st">"ALB"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">test_data</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"PDB_ID"</span>, <span class="st">"Chain"</span>, <span class="st">"Position"</span>, <span class="st">"Orig_AA"</span>, <span class="st">"Mut_AA"</span>, <span class="st">"Gene_Name"</span><span class="op">)</span></span>
<span><span class="va">KEGG_path_number</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/kegg_pathway_number.html">KEGG_pathway_number</a></span><span class="op">(</span>filtered_info_df <span class="op">=</span> <span class="va">test_data</span><span class="op">)</span> </span></code></pre></div>
</div>
</div>
<div class="section level4">
<h4 id="genic_intolerance">
<code>genic_intolerance</code><a class="anchor" aria-label="anchor" href="#genic_intolerance"></a>
</h4>
<p><code><a href="../reference/genic_intolerance.html">genic_intolerance()</a></code> function returns Genic Intolerance score by using a dataset retrieved from <a href="http://genic-intolerance.org/" class="external-link">Genic Intolerance</a>.</p>
<div class="section level5">
<h5 id="arguments-12">Arguments<a class="anchor" aria-label="anchor" href="#arguments-12"></a>
</h5>
<p><code><a href="../reference/genic_intolerance.html">genic_intolerance()</a></code> has 1 arguments;</p>
<ul>
<li>
<strong>filtered_info_df:</strong> input data.frame contains only one PDB entries</li>
</ul>
</div>
<div class="section level5">
<h5 id="usage-11">Usage<a class="anchor" aria-label="anchor" href="#usage-11"></a>
</h5>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">test_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">rbind</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"3SQJ"</span>, <span class="st">"A"</span>, <span class="fl">3</span>, <span class="st">"HIS"</span>, <span class="st">"GLN"</span>, <span class="st">"ALB"</span><span class="op">)</span>,</span>
<span>                                 <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"3SQJ"</span>, <span class="st">"A"</span>, <span class="fl">60</span>, <span class="st">"GLU"</span>, <span class="st">"LYS"</span>,<span class="st">"ALB"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">test_data</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"PDB_ID"</span>, <span class="st">"Chain"</span>, <span class="st">"Position"</span>, <span class="st">"Orig_AA"</span>, <span class="st">"Mut_AA"</span>, <span class="st">"Gene_Name"</span><span class="op">)</span></span>
<span><span class="va">genic_intolerance_score</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/genic_intolerance.html">genic_intolerance</a></span><span class="op">(</span>filtered_info_df <span class="op">=</span> <span class="va">test_data</span><span class="op">)</span> </span></code></pre></div>
</div>
</div>
<div class="section level4">
<h4 id="go_terms">
<code>GO_terms</code><a class="anchor" aria-label="anchor" href="#go_terms"></a>
</h4>
<p><code><a href="../reference/GO_terms.html">GO_terms()</a></code> function returns the number of GO terms related with the input genes.</p>
<div class="section level5">
<h5 id="arguments-13">Arguments<a class="anchor" aria-label="anchor" href="#arguments-13"></a>
</h5>
<p><code><a href="../reference/GO_terms.html">GO_terms()</a></code> has 1 arguments;</p>
<ul>
<li>
<strong>filtered_info_df:</strong> input data.frame contains only one PDB entries</li>
</ul>
</div>
<div class="section level5">
<h5 id="usage-12">Usage<a class="anchor" aria-label="anchor" href="#usage-12"></a>
</h5>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">test_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">rbind</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"3SQJ"</span>, <span class="st">"A"</span>, <span class="fl">3</span>, <span class="st">"HIS"</span>, <span class="st">"GLN"</span>, <span class="st">"ALB"</span><span class="op">)</span>,</span>
<span>                                 <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"3SQJ"</span>, <span class="st">"A"</span>, <span class="fl">60</span>, <span class="st">"GLU"</span>, <span class="st">"LYS"</span>,<span class="st">"ALB"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">test_data</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"PDB_ID"</span>, <span class="st">"Chain"</span>, <span class="st">"Position"</span>, <span class="st">"Orig_AA"</span>, <span class="st">"Mut_AA"</span>, <span class="st">"Gene_Name"</span><span class="op">)</span></span>
<span><span class="va">GO_terms_score</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/GO_terms.html">GO_terms</a></span><span class="op">(</span>filtered_info_df <span class="op">=</span> <span class="va">test_data</span><span class="op">)</span> </span></code></pre></div>
</div>
</div>
<div class="section level4">
<h4 id="disgenet">
<code>DisGeNET</code><a class="anchor" aria-label="anchor" href="#disgenet"></a>
</h4>
<p><code><a href="../reference/DisGeNET.html">DisGeNET()</a></code> function returns number of diseases related with input genes by using a dataset retrieved from <a href="http://https://www.disgenet.org/" class="external-link">DisGeNET</a>.</p>
<div class="section level5">
<h5 id="arguments-14">Arguments<a class="anchor" aria-label="anchor" href="#arguments-14"></a>
</h5>
<p><code><a href="../reference/DisGeNET.html">DisGeNET()</a></code> has 1 arguments;</p>
<ul>
<li>
<strong>filtered_info_df:</strong> input data.frame contains only one PDB entries</li>
</ul>
</div>
<div class="section level5">
<h5 id="usage-13">Usage<a class="anchor" aria-label="anchor" href="#usage-13"></a>
</h5>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">test_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">rbind</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"3SQJ"</span>, <span class="st">"A"</span>, <span class="fl">3</span>, <span class="st">"HIS"</span>, <span class="st">"GLN"</span>, <span class="st">"ALB"</span><span class="op">)</span>,</span>
<span>                                 <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"3SQJ"</span>, <span class="st">"A"</span>, <span class="fl">60</span>, <span class="st">"GLU"</span>, <span class="st">"LYS"</span>,<span class="st">"ALB"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">test_data</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"PDB_ID"</span>, <span class="st">"Chain"</span>, <span class="st">"Position"</span>, <span class="st">"Orig_AA"</span>, <span class="st">"Mut_AA"</span>, <span class="st">"Gene_Name"</span><span class="op">)</span></span>
<span><span class="va">DisGeNET_score</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/DisGeNET.html">DisGeNET</a></span><span class="op">(</span>filtered_info_df <span class="op">=</span> <span class="va">test_data</span><span class="op">)</span> </span></code></pre></div>
</div>
</div>
<div class="section level4">
<h4 id="gene_essentiality">
<code>gene_essentiality</code><a class="anchor" aria-label="anchor" href="#gene_essentiality"></a>
</h4>
<p><code><a href="../reference/gene_essentiality.html">gene_essentiality()</a></code> function returns Gene Essentiality scores of the input genes by using a dataset retrieved from <a href="https://v3.ogee.info/#/home" class="external-link">Online Gene Essentiality (OGEE) Database</a>.</p>
<div class="section level5">
<h5 id="arguments-15">Arguments<a class="anchor" aria-label="anchor" href="#arguments-15"></a>
</h5>
<p><code><a href="../reference/gene_essentiality.html">gene_essentiality()</a></code> has 1 arguments;</p>
<ul>
<li>
<strong>filtered_info_df:</strong> input data.frame contains only one PDB entries</li>
</ul>
</div>
<div class="section level5">
<h5 id="usage-14">Usage<a class="anchor" aria-label="anchor" href="#usage-14"></a>
</h5>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">test_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">rbind</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"3SQJ"</span>, <span class="st">"A"</span>, <span class="fl">3</span>, <span class="st">"HIS"</span>, <span class="st">"GLN"</span>, <span class="st">"ALB"</span><span class="op">)</span>,</span>
<span>                                 <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"3SQJ"</span>, <span class="st">"A"</span>, <span class="fl">60</span>, <span class="st">"GLU"</span>, <span class="st">"LYS"</span>,<span class="st">"ALB"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">test_data</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"PDB_ID"</span>, <span class="st">"Chain"</span>, <span class="st">"Position"</span>, <span class="st">"Orig_AA"</span>, <span class="st">"Mut_AA"</span>, <span class="st">"Gene_Name"</span><span class="op">)</span></span>
<span><span class="va">gene_essentiality_score</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/gene_essentiality.html">gene_essentiality</a></span><span class="op">(</span>filtered_info_df <span class="op">=</span> <span class="va">test_data</span><span class="op">)</span> </span></code></pre></div>
</div>
</div>
<div class="section level4">
<h4 id="gtex">
<code>GTEx</code><a class="anchor" aria-label="anchor" href="#gtex"></a>
</h4>
<p><code><a href="../reference/GTEx.html">GTEx()</a></code> function calculates the median gene expression value with using 54 different tissue types’ median gene expression value from <a href="https://gtexportal.org/home/" class="external-link">GTEx</a> and returns the value of input gene.</p>
<div class="section level5">
<h5 id="arguments-16">Arguments<a class="anchor" aria-label="anchor" href="#arguments-16"></a>
</h5>
<p><code><a href="../reference/GTEx.html">GTEx()</a></code> has 1 arguments;</p>
<ul>
<li>
<strong>filtered_info_df:</strong> input data.frame contains only one PDB entries</li>
</ul>
</div>
<div class="section level5">
<h5 id="usage-15">Usage<a class="anchor" aria-label="anchor" href="#usage-15"></a>
</h5>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">test_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">rbind</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"3SQJ"</span>, <span class="st">"A"</span>, <span class="fl">3</span>, <span class="st">"HIS"</span>, <span class="st">"GLN"</span>, <span class="st">"ALB"</span><span class="op">)</span>,</span>
<span>                                 <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"3SQJ"</span>, <span class="st">"A"</span>, <span class="fl">60</span>, <span class="st">"GLU"</span>, <span class="st">"LYS"</span>,<span class="st">"ALB"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">test_data</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"PDB_ID"</span>, <span class="st">"Chain"</span>, <span class="st">"Position"</span>, <span class="st">"Orig_AA"</span>, <span class="st">"Mut_AA"</span>, <span class="st">"Gene_Name"</span><span class="op">)</span></span>
<span><span class="va">gene_essentiality_score</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/GTEx.html">GTEx</a></span><span class="op">(</span>filtered_info_df <span class="op">=</span> <span class="va">test_data</span><span class="op">)</span> </span></code></pre></div>
</div>
</div>
<div class="section level4">
<h4 id="amino_acid_features">
<code>amino_acid_features</code><a class="anchor" aria-label="anchor" href="#amino_acid_features"></a>
</h4>
<p><code><a href="../reference/amino_acid_features.html">amino_acid_features()</a></code> function returns 6 different features:</p>
<ul>
<li>Accessible Surface Area of reference amino acid</li>
<li>Accessible Surface Area of mutant amino acid</li>
<li>Accessible Surface Area difference of mutant and reference amino acid</li>
<li>Hydrophobicity Scale of reference amino acid</li>
<li>Hydrophobicity Scale of mutant amino acid</li>
<li>Hydrophobicity Scale difference of mutant and reference amino acid</li>
</ul>
<div class="section level5">
<h5 id="arguments-17">Arguments<a class="anchor" aria-label="anchor" href="#arguments-17"></a>
</h5>
<p><code><a href="../reference/amino_acid_features.html">amino_acid_features()</a></code> has 1 arguments;</p>
<ul>
<li>
<strong>info_df:</strong> input data.frame contains all the mutations</li>
</ul>
</div>
<div class="section level5">
<h5 id="usage-16">Usage<a class="anchor" aria-label="anchor" href="#usage-16"></a>
</h5>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">test_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">rbind</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"3SQJ"</span>, <span class="st">"A"</span>, <span class="fl">3</span>, <span class="st">"HIS"</span>, <span class="st">"GLN"</span>, <span class="st">"ALB"</span><span class="op">)</span>,</span>
<span>                                 <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"3SQJ"</span>, <span class="st">"A"</span>, <span class="fl">60</span>, <span class="st">"GLU"</span>, <span class="st">"LYS"</span>,<span class="st">"ALB"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">test_data</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"PDB_ID"</span>, <span class="st">"Chain"</span>, <span class="st">"Position"</span>, <span class="st">"Orig_AA"</span>, <span class="st">"Mut_AA"</span>, <span class="st">"Gene_Name"</span><span class="op">)</span></span>
<span><span class="va">gene_essentiality_score</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/amino_acid_features.html">amino_acid_features</a></span><span class="op">(</span>info_df <span class="op">=</span> <span class="va">test_data</span><span class="op">)</span> </span></code></pre></div>
</div>
</div>
<div class="section level4">
<h4 id="impact_prediction">
<code>impact_prediction</code><a class="anchor" aria-label="anchor" href="#impact_prediction"></a>
</h4>
<p><code><a href="../reference/impact_prediction.html">impact_prediction()</a></code> function makes impact prediction and classifies the mutation as <strong>Pathogenic</strong> or <strong>Neutral</strong>.</p>
<div class="section level5">
<h5 id="arguments-18">Arguments<a class="anchor" aria-label="anchor" href="#arguments-18"></a>
</h5>
<p><code><a href="../reference/impact_prediction.html">impact_prediction()</a></code> has 3 arguments;</p>
<ul>
<li>
<strong>final_df:</strong> data.frame contains all the required information for impact prediction</li>
<li>
<strong>distance_cutoff:</strong> distance cutoff for setting edges (default = 7)</li>
<li>
<strong>network_approach:</strong> network building approach; “all” (default) for using all atoms or “ca” for using Cα atoms only</li>
</ul>
</div>
<div class="section level5">
<h5 id="usage-17">Usage<a class="anchor" aria-label="anchor" href="#usage-17"></a>
</h5>
<p>From an input data.frame containing <strong>PDB_ID</strong>, <strong>Chain</strong>, <strong>Position</strong>, <strong>Orig_AA</strong>, <strong>Mut_AA</strong>, <strong>Gene_Name</strong>, <strong>degree_z_score</strong>, <strong>eigen_z_score</strong>, <strong>shortest_path_z</strong>, <strong>betwenness_scores_z</strong>, <strong>clique_z_score</strong>, <strong>pagerank_z_score</strong>, <strong>syn_z</strong>, <strong>mis_z</strong>, <strong>pLI</strong>, <strong>blosum62_scores</strong>, <strong>kegg_pathway_number</strong>, <strong>genic_intolerance</strong>, <strong>go_terms</strong>, <strong>disgenet</strong>, <strong>gene_essentiality</strong>, <strong>gtex</strong>, <strong>ref_asa</strong>, <strong>mut_asa</strong>, <strong>asa_diff</strong>, <strong>ref_hyd</strong>, <strong>mut_hyd</strong> and <strong>hyd_diff</strong> scores <code><a href="../reference/impact_prediction.html">impact_prediction()</a></code> function classifies the mutation as <strong>Pathogenic</strong> or <strong>Neutral</strong>.</p>
<p>Example input:</p>
<div id="htmlwidget-89f231758d26bc202809" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-89f231758d26bc202809">{"x":{"filter":"none","vertical":false,"data":[["1","2"],["3SQJ","3SQJ"],["A","A"],["196","396"],["GLN","GLU"],["LEU","LYS"],["ALB","ALB"],["-0.0717218071350537","-0.280119006995483"],["-0.379784447875971","-0.406648724044331"],["-1.61712172099028","-0.114702339705536"],["0.709536957153219","-0.531050971889894"],["-0.0906809533248494","0.0706892006523297"],["0.0255568093251824","-0.318847672527394"],["-0.91513","-0.91513"],["0.65574","0.65574"],["0.64329","0.64329"],["-2","1"],["12","12"],["-0.89","-0.89"],["4","4"],["81","81"],["1","1"],["1.515105","1.515105"],["177.7","182.9"],["173.7","215.2"],["-4","32.3"],["-3.68","-3.2"],["3.8","-4.11"],["7.48","-0.91"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>PDB_ID<\/th>\n      <th>Chain<\/th>\n      <th>Position<\/th>\n      <th>Orig_AA<\/th>\n      <th>Mut_AA<\/th>\n      <th>Gene_Name<\/th>\n      <th>degree_z_score<\/th>\n      <th>eigen_z_score<\/th>\n      <th>shortest_path_z<\/th>\n      <th>betweenness_scores_z<\/th>\n      <th>clique_z_score<\/th>\n      <th>pagerank_z_score<\/th>\n      <th>syn_z<\/th>\n      <th>mis_z<\/th>\n      <th>pLI<\/th>\n      <th>blosum62_scores<\/th>\n      <th>kegg_pathway_number<\/th>\n      <th>genic_intolerance<\/th>\n      <th>go_terms<\/th>\n      <th>disgenet<\/th>\n      <th>gene_essentiality<\/th>\n      <th>gtex<\/th>\n      <th>ref_asa<\/th>\n      <th>mut_asa<\/th>\n      <th>asa_diff<\/th>\n      <th>ref_hyd<\/th>\n      <th>mut_hyd<\/th>\n      <th>hyd_diff<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"scrollX":true,"columnDefs":[{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script><div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">prediction_result</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/impact_prediction.html">impact_prediction</a></span><span class="op">(</span><span class="va">final_df</span><span class="op">)</span> </span></code></pre></div>
<table class="table">
<thead><tr class="header">
<th align="center">PDB_ID</th>
<th align="center">Chain</th>
<th align="center">Position</th>
<th align="center">Orig_AA</th>
<th align="center">Mut_AA</th>
<th align="center">Gene_Name</th>
<th align="center">Prediction</th>
<th align="center">Probability</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="center">3SQJ</td>
<td align="center">A</td>
<td align="center">196</td>
<td align="center">GLN</td>
<td align="center">LEU</td>
<td align="center">ALB</td>
<td align="center">Neutral</td>
<td align="center">0.6521832</td>
</tr>
<tr class="even">
<td align="center">3SQJ</td>
<td align="center">A</td>
<td align="center">396</td>
<td align="center">GLU</td>
<td align="center">LYS</td>
<td align="center">ALB</td>
<td align="center">Neutral</td>
<td align="center">0.6009792</td>
</tr>
</tbody>
</table>
<p><br><br></p>
</div>
</div>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Berk Gurdamar.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
